src/Action.ts

```diff
@@ -79,41 +83,52 @@ export class ActionBase {
   // safety for those places which should require it.
   // TODO: Remove ScoredAction since it doesn't have payload
   static GetPayload(action: ActionBase | ScoredBase, entityValues: Map<string, string>): string {
-    if (action.actionType === ActionTypes.TEXT) {
-      /**
-       * For backwards compatibility check if payload is of new TextPayload type
-       * Ideally we would implement schema refactor:
-       * 1. Make payloads discriminated unions (E.g. After checking the action.type, flow control knows the type of the payload property)
-       * This removes the need for the GetPayload function and GetArguments which are brittle coding patterns.
-       */
-      try {
-        const textPayload = JSON.parse(action.payload) as TextPayload
-        return EntityIdSerializer.serialize(textPayload.json, entityValues)
-      } catch (e) {
-        const error = e as Error
-        throw new Error(
-          `Error when attempting to parse text action payload. This might be an old action which was saved as a string.  Please create a new action. ${
-          error.message
-          }`
-        )
-      }
-    } else if (action.actionType === ActionTypes.END_SESSION) {
-      const textPayload = JSON.parse(action.payload) as TextPayload
-      return EntityIdSerializer.serialize(textPayload.json, entityValues)
-    }
-    // For API or CARD the payload field of the outer payload is the name of API or the filename of the card template without extension
-    else if (ActionTypes.CARD === action.actionType) {
-      let cardPayload = JSON.parse(action.payload) as CardPayload
-      return cardPayload.payload
-    } else if (ActionTypes.API_LOCAL === action.actionType) {
-      let actionPayload = JSON.parse(action.payload) as ActionPayload
-      return actionPayload.payload
-    } else if (ActionTypes.DISPATCH === action.actionType) {
-      // TODO: Another reason to schema refactor...
-      let actionPayload = JSON.parse(action.payload) as DispatchPayload
-      return `${ActionTypes.DISPATCH}: ${actionPayload.modelName}`
+    switch (action.actionType) {
+      case ActionTypes.TEXT:
+        {
```

Comment:
nit: there is no official typescript style guideline and not sure what tslint enforces here but prefer to have `{` in the same line as case statement to make it similar to usage of `{` in other constructs. 

```javascript 
 case ActionTypes.TEXT: {
       /**
       * For backwards compatibility check if payload is of new TextPayload type	   
```

Reply:
Done